File: Java/Math/divisors of a number.java

            //todos los divisores de un numero en O(sqrt(n))
            List<Long> div = new ArrayList<>();
            for (long i = 1; i * i <= n; i++) {
                if (n % i == 0) {
                    if (i * i == n) {
                        div.add(i);
                    } else {
                        div.add(i);
                        if (i != 1) {
                            div.add(n / i);
                        }
                    }
                }
            }
            

File: cpp/Math/Divisors of a number.cpp
//todos los divisores n en O(sqrt(n))
vll divisors(ll n) {
    vll div;
    for (ll i = 1; i * i <= n; i++) {
        if (n % i == 0) {
            div.pb(i);
            if ((n/i) != i)
                div.pb(n / i);
        }
    }
    return div;
}
